tree.go

```diff
@@ -277,49 +290,48 @@ func (n *Tree[K, V]) Size() int {
 }
 
 //LeastUpperBound returns the key-value-pair for the smallest node n such that n.Key() >= key. If there is no such
-//node then an empty pair is returned.
-func (n *Tree[K, V]) LeastUpperBound(key K) Pair[K, V] {
+//node then boolean is false.
+func (n *Tree[K, V]) LeastUpperBound(key K) (Pair[K, V], bool) {
 	if n.IsEmpty() {
-		return n
+		return nil, false
```

Comment:
Please consider `return n, false`.

this will return a default constructed pair instead of a nil interface and will reflect what happens when you make pair a struct. It will also simplify Set some as well.